hpedockerplugin/hpe/hpe_3par_common.py

```diff
@@ -970,6 +970,24 @@ def _get_prioritized_host_on_3par(self, host, hosts, hostname):
 
         return host, hostname
 
+    def revert_snap_to_vol(self, volume, snapshot):
+        try:
+            optional = {}
+            snapshot_name = self._get_3par_snap_name(snapshot['id'])
+            volume_name = self._get_3par_vol_name(volume['id'])
+            if self.client.isOnlinePhysicalCopy(volume_name):
+                LOG.info("Found an online copy for %(volume)s. ",
+                         {'volume': volume_name})
+                optional['online'] = True            
+            self.client.promoteVirtualCopy(snapshot_name, optional=optional)
+            LOG.info("Volume %(volume)s successfully reverted to"
+                     " %(snapname)s.", {'volume': volume_name,
+                                     'snapname': snapshot_name})
+        except Exception as ex:
+            LOG.error(_LE("Exception: %s"), ex)
+            raise exception.PluginException(ex)
```

Comment:
I would suggest raising more specific exception here like PromoteSnapshotException. This will help in the hpe_storage_api module to log proper message. As of now, if the operation fails here, we are logging ETCD lock related message in hpe_storage_api.

Reply:
okay